sql/analyzer/pushdown.go

```diff
@@ -420,6 +421,34 @@ func convertFiltersToIndexedAccess(
 	return node, nil
 }
 
+// pushdownFiltersToTable attempts to push down filters to indexes that can accept them.
+func pushdownFiltersToIndex(ctx *sql.Context, a *Analyzer, idxTable *plan.IndexedTableAccess, lookup *indexLookup, tableAliases TableAliases) (handled []sql.Expression, err error) {
+	filteredIdx, ok := idxTable.Index().(sql.FilteredIndex)
+	if !ok {
+		return nil, nil
+	}
+
+	idxFilters := splitConjunction(lookup.expr)
+	if len(idxFilters) == 0 {
+		return nil, nil
+	}
+	idxFilters = normalizeExpressions(ctx, tableAliases, idxFilters...)
+
+	handled = filteredIdx.HandledFilters(idxFilters)
+	if len(handled) == 0 {
+		return nil, nil
+	}
+
+	a.Log(
```

Comment:
might be useful to log more info, like the lookup expr when it's non-nil and the handled subset. if the two don't overlap it will stick out in debug logs. would have saved us both some trouble.